hook.js

```diff
@@ -107,12 +116,16 @@ function createHook (meta) {
     ) {
       return url
     }
-
+    
+    // on Node 16.12.0, url.format is undefined for .mjs files, so explicitly set it to 'module'
+    if (url.format === undefined && EXTENSION_MJS_RE.test(url.url)) {
```

Comment:
You might also need to check if `"type": "module"` in package.json. Actually, you might not. Can you clarify the comment? What are the conditions under which it's undefined, exactly? The documentation doesn't seem to indicate this.

Reply:
on Node's 16.0.0-16.12.0, url.format is undefined for the cyclical dependency test files ./test/fixtures/a.mjs & ./test/fixtures/b.mjs 
so this explicitly sets format to 'module' for files with a .mjs extension so that they can go through the ast parsing patch for Node >= 16

Reply:
Okay, what happens when `"type": "module"` and the file extension is just `.js`, but otherwise it's the same test/example?